proxy.php

```diff
@@ -407,18 +413,27 @@ function startsWith($string, $with) {
     } elseif ($parsingHeaders) {
```

Comment:
OK .. so loop-exit code _was_  just based on read error (beyond EOS or so) . Now content-length is being used as well. Good.